fontbe/src/features.rs

```diff
@@ -72,12 +76,157 @@ impl Display for NotSupportedError {
     }
 }
 
+// TODO: ask Colin if we need to be threadsafe
+struct FeaVariationInfo<'a> {
+    fea_rs_axes: HashMap<Tag, (&'a CoordConverter, fea_rs::compile::AxisInfo)>,
+    axes: HashMap<Tag, &'a Axis>,
+    static_metadata: &'a StaticMetadata,
+}
+
+impl<'a> FeaVariationInfo<'a> {
+    fn new(static_metadata: &'a StaticMetadata) -> FeaVariationInfo<'a> {
+        FeaVariationInfo {
+            fea_rs_axes: static_metadata
+                .variable_axes
+                .iter()
+                .enumerate()
+                .map(|(i, a)| {
+                    (
+                        a.tag,
+                        (
+                            &a.converter,
+                            fea_rs::compile::AxisInfo {
+                                index: i as u16,
+                                default_value: a.default.into(),
+                                min_value: a.min.into(),
+                                max_value: a.max.into(),
+                            },
+                        ),
+                    )
+                })
+                .collect(),
+            axes: static_metadata
+                .variable_axes
+                .iter()
+                .map(|a| (a.tag, a))
+                .collect(),
+            static_metadata,
+        }
+    }
+}
+
+impl<'a> VariationInfo for FeaVariationInfo<'a> {
+    fn axis_info(&self, axis_tag: font_types::Tag) -> Option<fea_rs::compile::AxisInfo> {
+        self.fea_rs_axes.get(&axis_tag).map(|a| a.1)
+    }
+
+    fn normalize_coordinate(
+        &self,
+        axis_tag: font_types::Tag,
+        value: font_types::Fixed,
+    ) -> font_types::F2Dot14 {
+        let converter = &self
+            .axes
+            .get(&axis_tag)
+            .expect("Unsupported axis")
+            .converter;
+        let user_coord = UserCoord::new(value.to_f64() as f32);
+        F2Dot14::from_f32(user_coord.to_normalized(converter).to_f32())
+    }
+
+    fn resolve_variable_metric(
+        &self,
+        values: &HashMap<BTreeMap<Tag, Fixed>, i16>,
+    ) -> Result<
+        (
+            i16,
+            Vec<(write_fonts::tables::variations::VariationRegion, i16)>,
+        ),
+        Box<(dyn StdError + 'static)>,
+    > {
+        // WARNING: this will fail if the fea location isn't also a glyph location. In time we may wish to fix that.
+        let var_model = &self.static_metadata.variation_model;
```

Comment:
I expect this reuse of a single VariationModel for all kerning pairs to create similar issues as in the sparse glyph variations #400 

The correct way to do this is to make specific VariationModels for each set of locations, fonttools does that and caches the models for the paris referencing the same set of locations, but am not sure if the latter optimization is needed for the Rust port. I think we should prefer correctness at first and only do the memoization if the profiler confirms it's worth it.

Reply:
SG but I would like to defer this, will file an issue.

Reply:
https://github.com/googlefonts/fontc/issues/407